src/base.c

```diff
@@ -25,15 +25,17 @@ base_map(extent_hooks_t *extent_hooks, unsigned ind, size_t size) {
 	if (extent_hooks == &extent_hooks_default) {
 		addr = extent_alloc_mmap(NULL, size, PAGE, &zero, &commit);
 	} else {
+		pre_reentrancy(tsdn_tsd(tsdn));
```

Comment:
I did check this part. The only place we pass NULL tsdn to base is when initializing a0. After that we always have tsd.